crates/turbopack/src/condition.rs

```diff
@@ -1,13 +1,16 @@
+use async_recursion::async_recursion;
+use futures::{stream, StreamExt};
 use serde::{Deserialize, Serialize};
 use turbo_tasks::trace::TraceRawVcs;
-use turbo_tasks_fs::FileSystemPath;
+use turbo_tasks_fs::{FileSystem, FileSystemKind, FileSystemPath};
 
 #[derive(Debug, Clone, Serialize, Deserialize, TraceRawVcs, PartialEq, Eq)]
 pub enum ContextCondition {
     All(Vec<ContextCondition>),
     Any(Vec<ContextCondition>),
     Not(Box<ContextCondition>),
     InDirectory(String),
+    Kind(FileSystemKind),
```

Comment:
I would rather use something like `InPath(FileSystemPathVc)` here since the fs type is more an implementation detail. It should be transparent to the user how the fs is implementated.

Reply:
The part I am not sure is how to distinguish internal path via given `FileSystemPathVc`. I think I may miss something, but path read by filesystempathvc doesn't give context if it is internal to the turbopack or if it's user's. Mind elaborate what did I miss?